src/java.base/share/classes/java/util/jar/JarInputStream.java

```diff
@@ -103,11 +148,13 @@ private JarEntry checkManifest(JarEntry e)
     }
 
     /**
-     * Returns the {@code Manifest} for this JAR file, or
-     * {@code null} if none.
+     * Returns the {@code Manifest} for this JAR file when it is the first entry
+     * in the stream (or the second entry the first entry in the stream is
+     * {@code META-INF/} and the second entry {@code META-INF/MANIFEST.MF}), or
+     * {@code null} otherwise.
      *
-     * @return the {@code Manifest} for this JAR file, or
-     *         {@code null} if none.
+     * @return the {@code Manifest} for this JAR file when accessible, or
+     *         {@code null} otherwise.
```

Comment:
The word "accessible" suggests there is access control in the picture so I think drop that word. Maybe just drop "if none" from the original return description?

Reply:
Will change as you suggest